Main/Controllers/CollectionsController.cs

```diff
@@ -47,5 +47,64 @@ public IActionResult AddCollection([FromBody] CollectionCreationData data)
       _dbContext.SaveChanges();
       return Ok();
     }
+
+    [HttpDelete("{id}")]
+    public IActionResult DeleteCollection(int id)
+    {
+      var userId = _userProvider.GetCurrentUserId();
+      var collection = _dbContext.Collection.SingleOrDefault(c => c.Id == id);
+      if (collection == null)
+        return NotFound();
+      if (collection.OwnerId == userId)
+      {
+        using (var dbContextTransaction = _dbContext.Database.BeginTransaction())
+        {
+          //delete connected saved collections
+          var savedCollections = _dbContext.SavedCollection.Where(sc => sc.CollectionId == id).ToArray();
+          _dbContext.RemoveRange(savedCollections);
+
+          //delete connected shared collections
+          var sharedCollections = _dbContext.SharedCollection.Where(sc => sc.CollectionId == id).ToArray();
+          _dbContext.RemoveRange(sharedCollections);
+
+          //delete collection's elements
+          var elements = _dbContext.Element.Where(sc => sc.CollectionId == id).ToArray();
+          _dbContext.RemoveRange(elements);
+
+          //delete collection
+          _dbContext.Remove(collection);
+
+          _dbContext.SaveChanges();
+          dbContextTransaction.Commit();
+        }
+        return Ok();
+      }
+      return Forbid();
+    }
+
+    [HttpPatch]
+    public IActionResult UpdateCollection([FromBody] CollectionUpdateData data)
+    {
+      var userId = _userProvider.GetCurrentUserId();
+      var collection = _dbContext.Collection.Single(c => c.Id == data.Id);
+      if (collection == null)
+        return NotFound();
+
+      var sharedCollection = _dbContext.SharedCollection.SingleOrDefault(sc => sc.CollectionId == data.Id && sc.UserId == userId);
+      bool canEditSharedCollection = sharedCollection == null 
+        ? false 
+        : sharedCollection.EditRights == null 
+          ? false 
+          : (bool)sharedCollection.EditRights;
```

Comment:
Never do something like this, it is not readable.
sharedCollection?.EditRights  == true should do the trick

Reply:
nice, didn't knew that I can do it like this